cpp/src/request_stream.cpp

```diff
@@ -41,7 +34,12 @@ std::shared_ptr<RequestStream> createRequestStream(std::shared_ptr<Endpoint> end
   auto req = std::shared_ptr<RequestStream>(
     new RequestStream(endpoint, send, buffer, length, enablePythonFuture));
 
-  req->_delayedSubmission->setParent(req);
+  // A delayed notification request is not populated immediately, instead it is
+  // delayed to allow the worker progress thread to set its status, and more
+  // importantly the Python future later on, so that we don't need the GIL here.
+  req->_worker->registerDelayedSubmission(
+    req, std::bind(std::mem_fn(&Request::populateDelayedSubmission), req.get()));
```

Comment:
Ah, this needs to be moved from the constructor because we need the `shared_ptr` to `this` which isn't available.